kodilla-hibernate/src/test/java/com/kodilla/hibernate/manytomany/dao/CompanyDaoTestSuite.java

```diff
@@ -59,4 +64,58 @@ public void testSaveManyToMany() {
         } catch (Exception e) {
         }
     }
+
+    @Test
+    public void testNamedQueries() {
+        //Given
+        Employee johnSmith = new Employee("John", "Smith");
+        Employee stephanieClarckson = new Employee("Stephanie", "Clarckson");
+        Employee lindaKovalsky = new Employee("Linda", "Kovalsky");
+        Employee marthaSmith = new Employee("Martha", "Smith");
+
+        Company softwareMachine = new Company("Software Machine");
+        Company dataMaesters = new Company("Data Maesters");
+        Company greyMatter = new Company("Grey Matter");
+        Company sofastic = new Company("Sofastic");
+
+        employeeDao.save(johnSmith);
+        int johnSmithId = johnSmith.getId();
+        employeeDao.save(stephanieClarckson);
+        int stephanieClarcksonId = stephanieClarckson.getId();
+        employeeDao.save(lindaKovalsky);
+        int lindaKovalskyId = lindaKovalsky.getId();
+        employeeDao.save(marthaSmith);
+        int marthaSmithId = marthaSmith.getId();
+        companyDao.save(softwareMachine);
+        int softwareMachineId = softwareMachine.getId();
+        companyDao.save(dataMaesters);
+        int dataMaestersId = dataMaesters.getId();
+        companyDao.save(greyMatter);
+        int greyMatterId = greyMatter.getId();
+        companyDao.save(sofastic);
+        int sofasticId = sofastic.getId();
+
+        //When
+        List<Employee> withGivenLastName = employeeDao.retrieveEmployeeWithGivenLastName("Smith");
+        List<Company> withNameBeginningWith = companyDao.retrieveCompaniesWithNameBeginningWith("Sof");
+
+        //Then
+        Assert.assertEquals(2, withGivenLastName.size());
+        Assert.assertEquals(2, withNameBeginningWith.size());
+        Assert.assertEquals("John", withGivenLastName.get(0).getFirstName());
```

Comment:
Tego rozwiązania nie można zaakceptować. Tak jak rozmawialiśmy - nie możesz polegać na konkretnej kolejności zwróconych z bazy rekordów (chyba że w zapytaniu jest `ORDER`).
Można to napisać na milion sposobów, problem jest jasno sprecyzowany: sprawdzenie, czy zbiór zawiera elementy o zadanych nazwach.
Jeśli nie chcesz robić tego przy pomocy pomocniczej metody, tak jak zaproponowałem, to możesz np. zamienić sobie `withNameBeginningWith` na listę `String`-ów (nazw) - przy pomocy `stream`, `map` i `collect` (to było dokładnie w module 7).